lib/matplotlib/axes/_axes.py

```diff
@@ -5287,9 +5287,10 @@ def fill_betweenx(self, y, x1, x2=0, where=None,
     #### plotting z(x, y): imshow, pcolor and relatives, contour
     @_preprocess_data()
     def imshow(self, X, cmap=None, norm=None, aspect=None,
-               interpolation=None, alpha=None, vmin=None, vmax=None,
-               origin=None, extent=None, *, filternorm=True, filterrad=4.0,
-               resample=None, url=None, **kwargs):
+               interpolation=None, interp_space=None, alpha=None,
```

Comment:
I don't think *anyone* is passing alpha positionally, so I guess the API break is safe, but in that case you may as well make all kwargs starting from `interp_space` keyword-only (as you effectively broke backcompat on passing `alpha` positionally).

Reply:
OK, not intentionally.  How far down the stack do we want to break it, if we had our druthers?  I'd probably put the `*` after `norm` and before `aspect`?

Reply:
Seems about right to me.  OTOH you can also just stick `interp_space` in the kwonly section for now and put a `@_api.make_keyword_only("3.5", "aspect")` and fix the order in 3.7.  No strong opinion there.

Reply:
I deprecated most of them!  (ahem, maybe we don't want this, but I really doubt too many people are using positional arguments for all of these)

Reply:
you only need to do it for the first one; the following ones implicitly become keyword-only as well (because you can't pass them positionally anymore without passing the first one positionally) :)

Reply:
ooops, duh!

Reply:
If I do this, I get an error on import
```
Traceback (most recent call last):
  File "/Users/jklymak/matplotlib/examples/images_contours_and_fields/image_antialiasing.py", line 20, in <module>
    import matplotlib.pyplot as plt
  File "/Users/jklymak/matplotlib/lib/matplotlib/pyplot.py", line 2875, in <module>
    def imshow(
  File "/Users/jklymak/matplotlib/lib/matplotlib/pyplot.py", line 101, in _copy_docstring_and_deprecators
    func = decorator(func)
  File "/Users/jklymak/matplotlib/lib/matplotlib/_api/deprecation.py", line 417, in make_keyword_only
    assert (name in signature.parameters
AssertionError: Matplotlib internal error: 'aspect' must be a positional-or-keyword parameter for imshow()
```

Reply:
Oops, I guess (I'm almost sure, actually) that make_keyword_only currently doesn't work with functions that have pyplot wrappers, because make_keyword_only fakes the displayed function signature to pretend that the switch to kwonly already occurred, but the pyplot wrapper should be generated using the old (non-kwonly) signature, but that's not the case right now.
It's going to be slightly annoying to fix but probably just needs some wrangling, can you open a separate issue to track this?  In any case I don't think this is required for this PR.